client/src/test/python/TestHttpClient.py

```diff
@@ -35,19 +40,41 @@ def setUp(self):
     def tearDown(self):
         pass
 
-    def test_get_with_oldstyle_cookie_string(self):
-        client = HttpClient()
-        client.verboseLevel = 0
-        client.cookie = 'acookie=foo=bar'
-        client.cookie_filename = '/dev/null'
+    def test_init_session_fail_no_creds(self):
+        ch = ConfigHolder()
+        ch.set('verboseLevel', 0)
+        ch.set('cookie_filename', '/dev/null')
 
-        client.init_session('http://foo.bar')
+        client = HttpClient(ch)
 
-        jar = RequestsCookieJar()
-        jar.update(client.session.cookies)
-        cookies = jar.get_dict(domain='foo.bar', path='/')
-        self.assertEqual(1, len(cookies))
-        self.assertEqual(cookies['acookie'], 'foo=bar')
+        self.assertRaisesRegexp(Exception, '^Unable to login',

```

Comment:
Test failing because of assertRaisesRegex but your removed the exception raise of unable to login from HttpClient. 
4 init_session test are failing